app/components/post_list/post/post.tsx

```diff
@@ -356,6 +358,10 @@ const Post = ({
                 joiningChannelId={null}
             />
         );
+    } else if (isUnrevealedPost) {
+        body = (
+            <UnrevealedBurnOnReadPost post={post}/>
```

Comment:
to avoid jumping up and down, perhaps we should look into having this as a mask of the content below until we find a way to scramble the contents

Reply:
The actual post content must never be sent to the client unless they reveal. This is a hard requirement. We discussed the post list jumping issue with UX as well and it was considered an acceptable compromise.